### PR TITLE
update usaco-669

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/silver/usaco-669.mdx [hunk 1]
```diff
@@ -7,9 +7,12 @@ author: Neo Wang, Óscar Garries, Kevin Sheng
 
 [Official Analysis (Java)](http://www.usaco.org/current/data/sol_moocast_gold_dec16.html)
 
-### Implementation (DSU)
+### Solution 1: DSU
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-669.mdx:10]: *implementation

File: solutions/silver/usaco-669.mdx [hunk 2]
```diff
@@ -7,9 +7,12 @@ author: Neo Wang, Óscar Garries, Kevin Sheng
...
+
+The following code describes the solution from the Official Analysis.
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-669.mdx:12]: official analysis doesn't have to be capitalized

File: solutions/silver/usaco-669.mdx [hunk 3]
```diff
@@ -7,9 +7,12 @@ author: Neo Wang, Óscar Garries, Kevin Sheng
...
 
 <LanguageSection>
+
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-669.mdx:15]:
```suggestion
```

File: solutions/silver/usaco-669.mdx [hunk 4]
```diff
@@ -252,38 +269,51 @@ print(last_dist, file=open("moocast.out", "w"))
...
+#include <queue>
 #include <vector>
 
 using std::cout;
 using std::endl;
 using std::pair;
+using std::queue;
 using std::vector;
 
+// Function to calculate the squared distance between two points
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-669.mdx:297]:
```suggestion
/** @return the squared distance between 2 points */
```

File: solutions/silver/usaco-669.mdx [hunk 5]
```diff
@@ -252,38 +269,51 @@ print(last_dist, file=open("moocast.out", "w"))
...
 int dist_sq(const pair<int, int> &p1, const pair<int, int> &p2) {
 	int dx = p1.first - p2.first;
 	int dy = p1.second - p2.second;
 	return dx * dx + dy * dy;
 }
 
+// Function to check if all cows are reachable given a certain power level
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-669.mdx:304]:
```suggestion
/** @return if all cows are reachable with the given power level */
```

**Review by SansPapyrus683 — APPROVED** [review]

**Comment by SansPapyrus683** [comment]
so uh here's the deal
either do these yourself or allow me to do them for you

File: solutions/silver/usaco-669.mdx [hunk 6]
```diff
@@ -9,6 +9,8 @@ author: Neo Wang, Óscar Garries, Kevin Sheng
 
 ### Implementation (DSU)
 
+The following code describes the solution from the official analysis.
+
```
Comment (SansPapyrus683) [inline solutions/silver/usaco-669.mdx:13]:
```suggestion
```

**Review by SansPapyrus683 — APPROVED** [review]
yeah

**Review by ryanchou-dev — APPROVED** [review]
amazing

**Pull request merged**